client/src/popups/AddClassificationModal.tsx

```diff
@@ -0,0 +1,506 @@
+  useEffect(() => {
+    const fetchClassificationOptions = async ({
+      query,
+      systemId,
+      categoryId,
+      subCategoryId,
+    }: {
+      query?: string;
+      systemId?: number;
+      categoryId?: number;
+      subCategoryId?: number;
+    }) => {
+      let searchParameters = "";
+      if (subCategoryId !== undefined) {
+        searchParameters = `subCategoryId=${subCategoryId}`;
+      } else if (categoryId !== undefined) {
+        searchParameters = `categoryId=${categoryId}`;
+      } else if (systemId !== undefined) {
+        searchParameters = `systemId=${systemId}`;
+      }
+
+      if (query !== undefined) {
+        if (searchParameters) {
+          searchParameters += "&";
+        }
+        searchParameters += `query=${query}`;
+      }
+
+      const results = await axios.get(
+        `/api/classifications/searchPossibleClassifications?${searchParameters}`,
+      );
+
+      const classifications: ContentClassification[] = results.data;
+      setClassificationOptions(classifications);
+    };
+    fetchClassificationOptions({ ...categoryFilter, query: queryFilter });
+  }, [categoryFilter, queryFilter]);
+
+  // Non-zero if waiting for server to add/remove classification (stores id)
+  const [dropdownWaitingForChange, setDropdownWaitingForChange] = useState(0);
+  useEffect(() => {
+    setDropdownWaitingForChange(0);
+  }, [existingClassifications]);
+
+  const selectedClassification =
+    categoryFilter.systemTreeIndex === undefined
+      ? null
+      : classificationCategories[categoryFilter.systemTreeIndex];
+
+  const classificationSystemOptionGroups: ReactElement[] = [];
+
+  let classificationTypeOptions: ReactElement[] = [];
+  let lastType = "";
+
+  for (const [i, system] of classificationCategories.entries()) {
+    if (system.type !== lastType && classificationTypeOptions.length > 0) {
+      if (lastType === "Primary") {
+        // skip optgroup for Primary
+        classificationSystemOptionGroups.push(...classificationTypeOptions);
+      } else {
+        classificationSystemOptionGroups.push(
+          <optgroup label={lastType} key={i}>
+            {classificationTypeOptions}
+          </optgroup>,
+        );
+      }
+      classificationTypeOptions = [];
+    }
+    lastType = system.type;
+
+    classificationTypeOptions.push(
+      <option value={i} key={i}>
+        {system.name}
+      </option>,
+    );
+
+    if (i === classificationCategories.length - 1) {
+      classificationSystemOptionGroups.push(
+        <optgroup label={system.type} key={i + 1}>
+          {classificationTypeOptions}
+        </optgroup>,
+      );
+    }
+  }
+
+  if (!isOpen) {
+    return null;
+  }
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} scrollBehavior="inside" size="2xl">
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>
+          <Heading size="md">Add a classification</Heading>
+        </ModalHeader>
+        <HStack>
```

Comment:
I believe these items should either be in `<ModalHeader>` or `<ModalBody>`. Since they are outside both, they display with no margin, and presumably would be hard to understand with assistive technology.